src/combinatorics/simple_graph/strongly_regular.lean

```diff
@@ -41,27 +40,133 @@ A graph is strongly regular with parameters `n k l m` if
  * every pair of adjacent vertices has `l` common neighbors
  * every pair of nonadjacent vertices has `m` common neighbors
 -/
-structure is_SRG_of (n k l m : ℕ) : Prop :=
+structure is_SRG_with (n k l m : ℕ) : Prop :=
 (card : fintype.card V = n)
 (regular : G.is_regular_of_degree k)
 (adj_common : ∀ (v w : V), G.adj v w → fintype.card (G.common_neighbors v w) = l)
-(nadj_common : ∀ (v w : V), ¬ G.adj v w ∧ v ≠ w → fintype.card (G.common_neighbors v w) = m)
+(nadj_common : ∀ (v w : V), v ≠ w ∧ ¬G.adj v w → fintype.card (G.common_neighbors v w) = m)
 
-open finset
+variables {n k l m : ℕ}
+
+/-- Empty graphs are strongly regular. Note that `l` can take any value
+  for empty graphs, since there are no pairs of adjacent vertices. -/
+lemma bot_strongly_regular :
+  (⊥ : simple_graph V).is_SRG_with (fintype.card V) 0 l 0 :=
+{ card := rfl,
+  regular := bot_degree,
+  adj_common := λ v w h, h.elim,
+  nadj_common := λ v w h, begin
+    simp only [finset.card_eq_zero, finset.filter_congr_decidable, fintype.card_of_finset],
+    ext,
+    simp [mem_common_neighbors],
+  end }
 
-/-- Complete graphs are strongly regular. Note that the parameter `m` can take any value
-  for complete graphs, since there are no distinct pairs of nonadjacent vertices. -/
-lemma complete_strongly_regular (m : ℕ) :
-  (⊤ : simple_graph V).is_SRG_of (fintype.card V) (fintype.card V - 1) (fintype.card V - 2) m :=
+/-- Complete graphs are strongly regular. Note that `m` can take any value
+  for complete graphs, since there are no distinct pairs of non-adjacent vertices. -/
+lemma top_strongly_regular :
```

Comment:
I'm at a loss how to properly name this. Yael would want `is_SRG_with_top` (well, he'd want `is_SRG_top` and rename `is_SRG_with` to `is_SRG`).

Keeping `is_SRG_with`, I think it's fair having the naming convention where the `is_` predicate goes to the end and we drop the preposition, so `top_is_SRG`.

However, maybe to avoid a big discussion, let's compromise with
```suggestion
lemma is_SRG_with.top :
```